modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/shp/MultiLineHandler.java

```diff
@@ -221,10 +221,13 @@ public Object read(ByteBuffer buffer, ShapeType type, boolean flatGeometry) {
                 }
             }
         }
-        if ((shapeType == ShapeType.ARCZ || shapeType == ShapeType.ARCM) && !flatGeometry) {
+        if ((doubleBuffer.hasRemaining() && (shapeType == ShapeType.ARCZ)
```

Comment:
Points and multipoints have not been covered by the changes, looking at their code, is seems they would be affected by the same issue. Or not?

Reply:
Multipoints also have the same problem. Points have already been fixed up in master by GEOT-6729 (commit 821d9add2f8de)